### PR TITLE
Tus for project dataset

### PR DESCRIPTION
**Pull request by klakhov**
### Motivation and context

Continue TUS integration to provide chunk upload for large dataset files in projects

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat-core/src/server-proxy.js [hunk 1]
```diff
@@ -612,41 +612,64 @@
...
+                        new FormData(), {
+                            params,
+                            proxy: config.proxy,
+                            headers: { 'Upload-Finish': true },
+                        });
+                    return new Promise((resolve, reject) => {
+                        async function requestStatus() {
+                            try {
+                                const response = await Axios.get(`${url}?action=import_status`, {
+                                    params,
```
Comment (bsekachev) [inline cvat-core/src/server-proxy.js:650]:
```suggestion
                                const response = await Axios.get(url, {
                                    params: { ...params, action: 'import_status' },
```

File: cvat-core/src/server-proxy.js [hunk 2]
```diff
@@ -612,41 +612,64 @@
...
+                                    proxy: config.proxy,
+                                });
+                                if (response.status === 202) {
+                                    if (onUpdate && response.data.message !== '') {
```
Comment (bsekachev) [inline cvat-core/src/server-proxy.js:654]:
```suggestion
                                    if (onUpdate && response.data.message) {
```

File: cvat-core/src/server-proxy.js [hunk 3]
```diff
@@ -612,41 +612,64 @@
...
-                    Axios.post(`${url}?format=${format}`, formData, {
-                        proxy: config.proxy,
-                    }).then(() => {
                         setTimeout(requestStatus, 2000);
-                    }).catch((error) => {
-                        reject(generateError(error));
                     });
-                });
+                } catch (errorData) {
+                    generateError(errorData);
```
Comment (bsekachev) [inline cvat-core/src/server-proxy.js:670]: Probably generated error should be raised
And we should not return null in this case.

**Comment by bsekachev** [comment]
Generally patch works for me except of 2 moments:

1. When I first time tried to import dataset, I get error with CSRF token (403), see gif above. I re-loginned and was not able to reproduce it anymore. So, maybe it was not related with this patch. Let's see if somebody meet the same issue. 
2. When error occurs, it does not show any errors, probably related with the error I commented in the code (generatedError is not raised)

error

File: cvat-core/src/server-proxy.js (hunk 3 above)
Comment (klakhov) [reply cvat-core/src/server-proxy.js:670]: Will such lines be better?
```
  generateError(errorData);
  throw errorData;
```
If we dont throw or return something, eslint `consistent-return` error occurs

Comment (bsekachev) [reply cvat-core/src/server-proxy.js:670]: Yep, looks like we do the same in other server methods.

File: cvat-core/src/server-proxy.js (hunk 1 above)
Comment (klakhov) [reply cvat-core/src/server-proxy.js]: Done

File: cvat-core/src/server-proxy.js (hunk 2 above)
Comment (klakhov) [reply cvat-core/src/server-proxy.js]: Done

**Review by nmanovic — APPROVED** [review]

**Pull request merged**